compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4041,15 +4066,61 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern, Location matc
                     return null;
             }
             return bLangVarBindingPattern;
-        } else if (matchPattern.kind() == SyntaxKind.LIST_MATCH_PATTERN) {
+        }
+
+        if (kind == SyntaxKind.ERROR_MATCH_PATTERN) {
+            ErrorMatchPatternNode errorMatchPatternNode = (ErrorMatchPatternNode) matchPattern;
+            BLangErrorMatchPattern bLangErrorMatchPattern =
+                    (BLangErrorMatchPattern) TreeBuilder.createErrorMatchPattern();
+            bLangErrorMatchPattern.pos = matchPatternPos;
+
+            NameReferenceNode nameReferenceNode;
+            if (errorMatchPatternNode.typeReference().isPresent()) {
+                nameReferenceNode = errorMatchPatternNode.typeReference().get();
+                bLangErrorMatchPattern.errorTypeReference = (BLangUserDefinedType) createTypeNode(nameReferenceNode);
+            }
+
+            Node node = errorMatchPatternNode.argListMatchPatternNode().get(0);
```

Comment:
The error-match-pattern can have no args, right? Note the second part of `error-arg-list-match-pattern`

```
error-match-pattern := error [error-type-reference] ( error-arg-list-match-pattern )
error-arg-list-match-pattern :=
   error-message-match-pattern [, error-cause-match-pattern] [, error-field-match-patterns]
   | [error-field-match-patterns]
```

The following results in a bad, sad error here.

```ballerina
function matchError(any|error e) {
    match e {
        error() => {
        }
    }
}
```

Reply:
Fixed